### PR TITLE
Added new reviews to profile page

### PR DESCRIPTION
**Pull request by tfaticone**
This enabled users to submit new reviews to a persons profile

### CODE REVIEW

File: src/components/PublicProfile.vue [hunk 1]
```diff
@@ -65,18 +135,45 @@
...
+      saveNewReview: function(){
+        let review = {
+          comment: this.additional_comments,
+          metrics: {
+            cleanliness: this.cleanliness,
+            loudness: this.loudness,
+            respectfulness: this.respectfulness,
+            sociability: this.sociability,
+          },
+          rater: this.user.displayName,
```
Comment (steplica) [inline src/components/PublicProfile.vue]: It looks like you forgot to update this 'this.user' reference to 'this.viewedUser'

File: src/components/PublicProfile.vue [hunk 2]
```diff
@@ -42,17 +102,27 @@
 <script>
   import db from '../database';
   import util from '../util/util';
+  import firebase from 'firebase';
```
Comment (steplica) [inline src/components/PublicProfile.vue]: It doesn't look like you need to import 'firebase'

File: src/components/PublicProfile.vue [hunk 3]
```diff
@@ -42,17 +102,27 @@
...
+  import AuthMixin from '../mixins/auth'
```
Comment (steplica) [inline src/components/PublicProfile.vue]: It doesn't look like you need to important the AuthMixin either?

File: src/components/PublicProfile.vue [hunk 4]
```diff
@@ -42,17 +102,27 @@
...
   const metricsRef = db.ref('metrics');
   const traitsRef = db.ref('traits');
   const schoolsRef = db.ref('schools');
 
   export default {
     name: 'profile',
+    mixins: [AuthMixin],
     data() {
       return {
+        traits: [],
```
Comment (steplica) [inline src/components/PublicProfile.vue]: I would give these new 'data' members more descriptive names.
'traits' really refers to 'newReviewTraits' right?
You'd also want to do the same for the other metrics:
'cleanliness' -> 'newReviewCleanliness'
'loudness' -> 'newReviewLoudness' etc.

File: src/components/PublicProfile.vue [hunk 5]
```diff
@@ -118,6 +215,22 @@
           overallRating: (overallRating / Object.keys(userMetrics).length).toFixed(1),
         };
       },
+      stepOneNewReview: function() {
```
Comment (steplica) [inline src/components/PublicProfile.vue:217]: I would change this to use object method shorthand:
stepOneNewReview() { ...
per Airbnb's JS style guide: https://github.com/airbnb/javascript#es6-object-shorthand

And I would change the two functions below as well.

File: src/components/PublicProfile.vue [hunk 6]
```diff
@@ -65,18 +135,45 @@
...
-        user: {
+        viewedUser: {
           source: schoolsRef.child(this.school).child(this.userId),
           asObject: true,
         },
       }
     },
+    methods: {
+      saveNewReview: function(){
+        let review = {
```
Comment (steplica) [inline src/components/PublicProfile.vue]: use 'const' over 'let' if the variable doesn't get reassigned.

File: src/components/PublicProfile.vue [hunk 7]
```diff
@@ -65,18 +135,45 @@
...
+          metrics: {
+            cleanliness: this.cleanliness,
+            loudness: this.loudness,
+            respectfulness: this.respectfulness,
+            sociability: this.sociability,
+          },
+          rater: this.user.displayName,
+          date: new Date().toLocaleString(),
+        };
+        let newTraits = {};
```
Comment (steplica) [inline src/components/PublicProfile.vue:156]: use 'const' over 'let'

File: src/components/PublicProfile.vue [hunk 8]
```diff
@@ -65,18 +135,45 @@
...
+        for(let trait in this.traits){
```
Comment (steplica) [inline src/components/PublicProfile.vue]: Prefer ES6 forEach https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach:
Array.forEach((item) => {
     // do things
});

File: src/components/PublicProfile.vue [hunk 9]
```diff
@@ -65,18 +135,45 @@
...
+          if(this.traits.hasOwnProperty(trait)){
+            newTraits[this.traits[trait]] = true;
+          }
+        }
+        review['traits'] = newTraits;
+        let ratingsRef = schoolsRef.child(this.school).child(this.userId).child('ratings');
```
Comment (steplica) [inline src/components/PublicProfile.vue]: use 'const' over 'let'

File: src/components/PublicProfile.vue [hunk 10]
```diff
@@ -65,18 +135,45 @@
...
+        let newRatingsRef = ratingsRef.push();
```
Comment (steplica) [inline src/components/PublicProfile.vue]: use 'const' over 'let'

File: src/components/PublicProfile.vue (hunk 1 above)
Comment (tfaticone) [reply src/components/PublicProfile.vue]: That is alright, it is referring to the user in AuthMixin

File: src/components/PublicProfile.vue (hunk 3 above)
Comment (tfaticone) [reply src/components/PublicProfile.vue]: WE do need this for current user

File: src/components/PublicProfile.vue (hunk 2 above)
Comment (tfaticone) [reply src/components/PublicProfile.vue]: Removed

**Comment by Justin-Nietzel** [comment]
I'm taking responsibility if this breaks anything.

**Pull request merged**